feat: Disable language analysis for a repository DOCS-686 (#2112)
codacy/docs: docs/repositories-configure/languages.md

---
description: Configure the languages that Codacy analyzes in your repository. You can configure Codacy to include unrecognized file extensions, or disable the analysis of specific languages.
---

# Configuring languages

{% include-markdown "../assets/includes/admin-access-control-info.md" %}

Codacy automatically detects and analyzes languages in your repository based on a list of supported file extensions. On the repository **Settings**, tab **Languages**, you can see the list of languages that Codacy detected and will include in the next analysis.

By default, the analysis is enabled for all languages that Codacy detects in you repository. You can adapt this configuration to your needs as follows:

-   [Configure Codacy to include unrecognized extensions](#configuring-file-extensions)
-   [Disable the analysis of specific languages](#disable-language)

!!! note

>     Codacy applies both **Languages** settings defined on the Codacy UI and any other language setting defined using a [Codacy configuration file](codacy-configuration-file.md#languages).

Author: Adding a tiny tweak to your suggestion:

```suggestion
    Codacy applies the language settings defined in the Codacy UI as well as any settings defined using a [Codacy configuration file](codacy-configuration-file.md#languages).
```
